sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TypeUtils.scala

```diff
@@ -73,4 +73,14 @@ object TypeUtils {
     }
     x.length - y.length
   }
+
+  /**
+   * Returns true if elements of the data type could be used as items of a hash set or as keys
```

Comment:
I think this comment is not very coherent with the method name. I think we can rephrase to something like:
```
Returns true if the equal method of the elements of the data type is implemented properly.
This also means that they can be safely used in collections relying on the equals method,
as sets or maps.
```
What do you think?

Reply:
I'm open to any changes :) But if you want to explicitly mention the ```equals``` method, I would also mention ```hashCode``` generally needed for usage in "hash" collections. But then this not 100% true for Spark's specialized ```OpenHashSets``` and ```OpenHashMaps``` since they calculate hash by themselves. WDYT?

Reply:
I think it is a common pattern for every well formed class to have equals and hashCode defined in a coherent way. I think what we are doing here is saying: this class has a meaningful equals method, so we can rely on it or this class has a meaningless equals method, like the default one comparing the pointers, so we cannot rely on it. I am open too to any suggestion, I'd only like to have a description which is coherent with the method name, otherwise I feel that either one or the other has to be changed in order to properly reflect what the method does.

Reply:
What about changing the name of the method to ```typeCanBeHashed```?

Reply:
I don't really like this proposal...